pkg/flow/internal/controller/component.go

```diff
@@ -219,6 +232,9 @@ func (cn *ComponentNode) ComponentName() string { return cn.componentName }
 // block.
 func (cn *ComponentNode) NodeID() string { return cn.nodeID }
 
+// GlobalNodeID returns the parent + nodeid of the component. It is guaranteed to be unique across all dags.
```

Comment:
I noticed in a lot of places we logged with `NodeID` which doesnt use the full path.